packages/react-composites/src/composites/CallWithChatComposite/ChatButton/useUnreadMessagesTracker.ts

```diff
@@ -42,14 +45,24 @@ export const useUnreadMessagesTracker = (chatAdapter: ChatAdapter, isChatPaneVis
       }
     };
 
-    chatAdapter.on('messageReceived', incrementUnreadChatMessagesCount);
-    chatAdapter.on('messageDeleted', decrementUnreadChatMessagesCount);
+    if (callWithChatAdapter.isChatAdapterInitialized()) {
+      console.log('ChatAdapter is already initialized');
+      callWithChatAdapter.on('messageReceived', incrementUnreadChatMessagesCount);
+      callWithChatAdapter.on('messageDeleted', decrementUnreadChatMessagesCount);
+    } else {
+      console.log('ChatAdapter is not initialized yet. Waiting for chatInitialized event.');
+      callWithChatAdapter.on('chatInitialized', (chatAdapter) => {
+        console.log('ChatAdapter is connected. Subscribing to messageReceived and messageDeleted events');
+        callWithChatAdapter.on('messageReceived', incrementUnreadChatMessagesCount);
+        callWithChatAdapter.on('messageDeleted', decrementUnreadChatMessagesCount);
+      });
+    }
 
     return () => {
-      chatAdapter.off('messageReceived', incrementUnreadChatMessagesCount);
-      chatAdapter.off('messageDeleted', decrementUnreadChatMessagesCount);
+      callWithChatAdapter.off('messageReceived', incrementUnreadChatMessagesCount);
+      callWithChatAdapter.off('messageDeleted', decrementUnreadChatMessagesCount);
     };
-  }, [chatAdapter, setUnreadChatMessages, isChatPaneVisible]);
+  }, [callWithChatAdapter, setUnreadChatMessages, isChatPaneVisible]);
```

Comment:
this useEffect won't be retriggered once `isChatAdapterInitialized`, instead, once isChatAdapterInitialized is moved to state (per other comment), we can write a selector where we have `const isChatInitialized = useSelector(isChatInitializedSelector)` that then watches the values of the isChatInitialized and will run any UI logic accordingly

Reply:
The alternative is to keep this as `chatAdapter` which we will only have when chat is available -- why did this switch from chatAdapter to callWithChatAdapter?